post/author.go

```diff
@@ -0,0 +1,123 @@
+package post
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+	log "github.com/sirupsen/logrus"
+)
+
+func (repository *Repository) GetPostsByAuthor(c *gin.Context) {
+	username := c.Param("username")
+
+	skip, err := strconv.Atoi(c.DefaultQuery("skip", "0"))
+	if err != nil {
+		log.Warn(err)
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	limit, err := strconv.Atoi(c.DefaultQuery("limit", "10"))
+	if err != nil {
+		log.Warn(err)
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	isPublished, err := strconv.ParseBool(c.DefaultQuery("is_published", "false"))
+	if err != nil {
+		log.Warn(err)
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	query := ""
+	if isPublished {
+		query = " WHERE p.is_published = true "
+	}
+
+	posts := []Post{}
+	err = repository.Db.Select(&posts, `SELECT p.id, p.title, p.slug, p.tags as tag, p.is_featured, p.is_published, p.published_on, p.published_at, p.reading_time, p.summary 
+										FROM posts p
+										JOIN posts_author_links pa ON p.id = pa.post_id
+										JOIN authors a ON a.id = pa.author_id AND a.username = $1`+query+`
+										ORDER BY p.published_on DESC
+										LIMIT $2 OFFSET $3`, username, limit, skip)
```

Comment:
Let's use ? instead of $, as it will not enforce to remember arg number.

Reply:
yaa I have tried but this is the syntax for postgres, we are not able to use (?)